File: Solution/CSharp/121.BestTimetoBuyandSellStock.cs
public class Solution {

    public int MaxProfit(int[] prices) {
        if (prices.Length == 0) {
            return 0;
        }
        int low = prices[0];
        int max = 0;
        int length = prices.Length;
        for (int i = 0; i < length; i++) {
            if (prices[i] < low) {
                low = prices[i];
            } else if (prices[i] - low > max) {
                max = prices[i] - low;
            }
        }
        return max;
    }
}

File: Solution/Python/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        # maxInt
        minPrice = 2**31 - 1
        maxProfit = 0
        for i in prices:
            minPrice = min(i, minPrice)
            maxProfit = max(i - minPrice, maxProfit)
        return maxProfit


s = Solution()
prices = [7, 1, 5, 3, 6, 4]
res = s.maxProfit(prices)
print(res)
